### PR TITLE
feat: Uplifted to PermissionsV2 schema.

### PR DESCRIPTION
**Pull request by TomBeckett**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

## Description of the pull request

I've moved the Roles and Permissions to match the new Permissions V2 schema/endpoints. 

Updated message uploads to not `enrichResponse` since server headers are not returned for those endpoints.

### CODE REVIEW

File: src/main/java/io/getstream/chat/java/models/Permission.java [hunk 1]
```diff
@@ -51,77 +54,22 @@
...
   @Builder(
       builderClassName = "PermissionCreateRequest",
       builderMethodName = "",
       buildMethodName = "internalBuild")
   public static class PermissionCreateRequestData {
+    @NotNull
+    @JsonProperty("id")
+    private String id;
+
     @Nullable
```
Comment (TomBeckett) [inline src/main/java/io/getstream/chat/java/models/Permission.java:66]: This should probably be `@NotNull` also

Comment (ferhatelmas) [reply src/main/java/io/getstream/chat/java/models/Permission.java:66]: That's correct. Could you push it too?

Comment (TomBeckett) [reply src/main/java/io/getstream/chat/java/models/Permission.java]: @ferhatelmas Pushed! Sorry for delay, I was kind of expecting some comments and was going to do it all in one pass 👍

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Comment by ferhatelmas** [comment]
@TomBeckett We might need to update some code examples in readme though

**Comment by TomBeckett** [comment]
@ferhatelmas Good point. Is that done as part of releasing or do you want a quick PR?

**Comment by ferhatelmas** [comment]
> do you want a quick PR

will appreciate if you could. Thanks!

**Pull request merged**